Lib/Protocols/IdHL7.pas

```diff
@@ -184,25 +186,30 @@   EHL7CommunicationError = class(EIdException)
   NULL_IP = '0.0.0.0';  {do not localize}
   DEFAULT_CONN_LIMIT = 1;
   DEFAULT_RECONNECT_DELAY = 15000;
+  DEFAULT_CONNECTION_TIMEOUT = 0;
   DEFAULT_COMM_MODE = cmUnknown;
   DEFAULT_IS_LISTENER = True;
   MILLISECOND_LENGTH = (1 / (24 * 60 * 60 * 1000));
+  SEND_RESPONSE_NAMES : Array [TSendResponse] Of String = ('None', 'Error', 'NoConnection', 'Sent', 'OK', 'Timeout');
 
 type
   // the connection is provided in these events so that applications can obtain information about the
   // the peer. It's never OK to write to these connections
-  TMessageArriveEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String) of object;
-  TMessageReceiveEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String; var VHandled: Boolean; var VReply: String) of object;
-  TReceiveErrorEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String; AException: Exception; var VReply: String; var VDropConnection: Boolean) of object;
+  TMessageArriveEvent = Procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: AnsiString) Of Object;
+  TMessageReceiveEvent = Procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: AnsiString; Var VHandled: Boolean; Var VReply: AnsiString) Of Object;
+  TReceiveErrorEvent = Procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: AnsiString; AException: Exception; Var VReply: AnsiString; Var VDropConnection: Boolean) Of Object;
 
   TIdHL7 = class;
   TIdHL7ConnCountEvent = procedure (ASender : TIdHL7; AConnCount : integer) of object;
+  TIdPeerThread = TIdContext;

```

Comment:
TIdPeerThread was an Indy 9 class, I'm not re-introducing it in Indy 10.  or Indy 10, please use TIdContext directly in methods as needed.

Reply:
Changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9